src/Microsoft.Health.Dicom.Core/Exceptions/ElementValidationException.cs

```diff
@@ -0,0 +1,47 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.Validation;
+
+namespace Microsoft.Health.Dicom.Core.Exceptions
+{
+    public class ElementValidationException : ValidationException
+    {
+        public ElementValidationException(string name, DicomVR vr, ValidationErrorCode errorCode, string message) : base(message)
+        {
+            Name = EnsureArg.IsNotNull(name, nameof(name));
+            VR = EnsureArg.IsNotNull(vr, nameof(vr));
+            ErrorCode = errorCode;
+        }
+
+        public ElementValidationException(string name, DicomVR vr, string value, ValidationErrorCode errorCode, string message) : base(message)
```

Comment:
Do we actually care if `Value` is `null` or not? If not, we could simplify the first ctor to something like:
```csharp
public ElementValidationException(string name, DicomVR vr, ValidationErrorCode errorCode, string message)
    : this(name, vr, null , errorCode, message)
{
}
```

Reply:
There are some cases we log value into error message, while other cases don't want. They distinguished by given value or not.
e.g: When DA value is invalid, we log tag value. When there are multiple values in one tag, we don't log.